### ISSUE TITLE
ATL-06: Memory is a separate module -- should be parameters [CS-01]

### ISSUE THREAD

**Issue by toddwbucy** [toddwbucy/Atlas-MAG_OmegaRule#55]
## Code Smell Violation: CS-01

**Rule**: No `MemoryModule` class (memory IS the parameters)
**Severity**: High (architectural alignment)

## Location

- `src/model/atlas_memory.py` -- `AtlasMemoryPoly` class (313 lines, standalone nn.Module)
- `src/model/blocks.py:193` -- `self.memory = AtlasMemoryPoly(...)` (composed as separate component)

## Impact

Memory is implemented as an independent `nn.Module` that is composed into the MAGBlock. In the NL ontology, **memory IS the parameters** -- there should not be a separate "memory module" that you can instantiate, compose, or swap independently.

The current design treats memory as a pluggable component:
```python
self.memory = AtlasMemoryPoly(dim=dim, key_dim=key_dim, ...)
```

But in the NL worldview, the memory weights (M matrices), the polynomial expansion, and the Q-K projection are all part of what the block IS, not something the block HAS.

## Fix Strategy

Large refactor -- integrate memory weights directly into the block as parameters:

1. Move `AtlasMemoryPoly` weight matrices (`W_q_mem`, `W_k_mem`, `M_memory`, etc.) into `MAGBlock` as direct `nn.Parameter`s
2. Move polynomial feature expansion into `MAGBlock.forward()` as inline computation
3. Move Q-K projection into `MAGBlock.forward()`
4. Remove `AtlasMemoryPoly` class entirely
5. Update all tests that reference `block.memory`

The forward pass computation stays identical -- only the ownership of parameters changes.

## Risk

**High** -- largest refactor of all graph compliance issues:
- 313 lines of `atlas_memory.py` absorbed into `blocks.py`
- All tests referencing `model.blocks[i].memory` must be updated
- Must verify parameter counts and gradient flow remain identical

## Graph Reference

- Code smell: `nl_code_smells/CS-01`
- NL IS #7: Self-modifying learning module (the module modifies its own parameters, not a separate memory object)
- NL IS NOT #4: NOT discrete long/short-term memory (no separate memory module)

## Migration Note

Identified during NL graph compliance audit (2026-02-09).
Tracked locally in `NestedLearning/Atlas/ISSUES.md` as ATL-06.
Defer to later pass -- this is the largest single refactor.
